src/dht.cpp

```diff
@@ -939,6 +944,12 @@ Dht::searchStep(Search& sr)
 
     /* Check if the first TARGET_NODES (8) live nodes have replied. */
     if (sr.isSynced(now)) {
+
+        if (sr.nodes.size()-sr.getNumberOfCandidates(now) < SEARCH_NODES and not sr.refilled) {
```

Comment:
Why restricting this to the synced state ? Synced means all 8 first known nodes have answered, so in that case there would be no need to refill at all.